### PR TITLE
feature/SC-4075-TeamCreationPermissions

### PR DESCRIPTION
**Pull request by mwolowyk-cap**
PR in Server: https://github.com/schul-cloud/schulcloud-server/pull/1269
Ticket: https://ticketsystem.schul-cloud.org/browse/SC-4075

### CODE REVIEW

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 86
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 86
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 86
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

File: config/default.schema.json [hunk 1]
```diff
@@ -194,6 +205,7 @@
 	},
 	"required": [
 		"TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION",
-		"COOKIE"
+		"COOKIE",
+		"STUDENT_TEAM_CREATION"
```
Comment (tofixx) [inline config/default.schema.json:209]: if setting this required, which is good, the default opt-in has to be added into the default.json too.

Comment (mwolowyk-cap) [reply config/default.schema.json:209]: Done

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 86
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 86
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 86
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

**Comment by mwolowyk-cap** [comment]
- Opt-out:

image

File: views/administration/teams.hbs [hunk 2]
```diff
@@ -10,24 +10,36 @@
         <script src="/scripts/administration/teams.js" type="text/javascript" defer></script>
     {{/content}}
     {{#content "page" mode="prepend"}}
-        <form action="/administration/teams/disablestudents/{{school._id}}" method="post" class="edit-form-school">
+		{{#ifConfig "STUDENT_TEAM_CREATION" "disabled"}}
```
Comment (tofixx) [inline views/administration/teams.hbs:13]: please add a default sentence that the users know about the setting and that it is disabled.

**Comment by mwolowyk-cap** [comment]
- Disabled: 
image


- Enabled:
image


- Opt-In:
image


- Opt-Out:
image

**Comment by mwolowyk-cap** [comment]
Opt-in (default)
image

Opt-Out (default)
image

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 86
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 36
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5488282&bid=17985912)

**Pull request merged**